lib/ctblfuns.gi

```diff
@@ -4749,50 +4742,21 @@ InstallGlobalFunction( ZevDataValue, function( q, n )
 #F  ZevData( <q>, <n> )
 #F  ZevData( <q>, <n>, <listofpairs> )
 ##
-InstallGlobalFunction( ZevData, function( arg )
-    local q,
-          n,
-          pos,
-          pos2;
-
-    q:= arg[1];
-    n:= arg[2];
-
-    pos:= Position( ZEV_DATA[1], q );
-    if pos = fail then
-      Add( ZEV_DATA[1], q );
-      Add( ZEV_DATA[2], [ [], [] ] );
-      pos:= Length( ZEV_DATA[1] );
-    fi;
-
-    pos2:= Position( ZEV_DATA[2][ pos ][1], n );
-    if pos2 = fail then
-      Add( ZEV_DATA[2][ pos ][1], n );
-      pos2:= Length( ZEV_DATA[2][ pos ][1] );
-    fi;
+InstallGlobalFunction( ZevData, function( q, n, listofpairs... )
+    local aux, result;
 
-    if Length( arg ) = 3 then
+    aux := GET_FROM_SORTED_CACHE(ZEV_DATA, q, {} -> NEW_SORTED_CACHE(false));
```

Comment:
Am I understanding this right, that `ZEV_DATA` itself caches other cashes? And if so, why is `ZEV_DATA` flushable but the caches contained in it are not?

Reply:
I've never noticed `ZEV_DATA` before I made this PR either, I only found it because I looked for places using `InstallFlushableValue`.

It is really a cache whose keys are pairs `(q,n)`, but it is implemented in this nested fashion, which I mapped to a "cache of caches". Depending on actual usage, it might be better to use a single cache with the pairs `[q,n]` as key, but I did not want to stray too far from what the code does right now...

The contained caches are not flushable because they are automatically flushed (i.e., discarded): when `ZEV_DATA` is flushed, its content is discarded, i.e. the caches it contained are removed; flushing them separately would just incur extra work with no benefit

Reply:
Thanks! That sounds sensible, I wanted to be sure that no other code assumes that the contents of `ZEV_DATA` is still there after flushing.